apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ServiceNameUtil.java

```diff
@@ -84,12 +89,14 @@ public static String replaceDisallowedChars(String serviceName) {
     @Nullable
     private static String parseJarCommand(String command) {
         final String[] commandParts = command.split(" ");
+        String result = null;
         for (String commandPart : commandParts) {
             if (commandPart.endsWith(".jar")) {
-                return removeVersionFromJar(removePath(removeJarExtension(commandPart)));
+                result = removeVersionFromJar(removePath(removeJarExtension(commandPart)));
+                break;
```

Comment:
using a break here allows to maximize coverage, otherwise the `return null;` statement was dead code